services/task-svc/internal/ward/ward.go

```diff
@@ -237,4 +231,73 @@ func (s ServiceServer) GetWardOverviews(ctx context.Context, _ *pb.GetWardOvervi
 	})
 
 	return &pb.GetWardOverviewsResponse{Wards: resWards}, err
+
+}
+
+func (ServiceServer) GetWardDetails(ctx context.Context, req *pb.GetWardDetailsRequest) (*pb.GetWardDetailsResponse, error) {
+
+	wardID, err := uuid.Parse(req.Id)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	wardRepository := wardmodels.NewWardRepositoryWithDB(hwgorm.GetDB(ctx))
+	ward, err := wardRepository.GetWardById(wardID)
+	if err != nil {
+		return nil, err
+	}
+
+	roomRepository := roommodels.NewRoomRepositoryWithDB(hwgorm.GetDB(ctx))
+	rooms, err := roomRepository.GetRoomByWard(wardID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	bedRepository := bedmodels.NewRoomRepositoryWithDB(hwgorm.GetDB(ctx))
+	var mappedRooms = hwutil.Map(rooms, func(room roommodels.Room) *pb.GetWardDetailsResponse_Room {
+		beds, _ := bedRepository.GetBedsByRoom(room.ID)
```

Comment:
Please do not ignore the error. Use `hwutil.MapWithErr` instead. Also happens further down below in this method.

Reply:
@florian-sabonchi this error gets still ignored :/